udata/core/dataset/models.py

```diff
@@ -638,6 +638,14 @@ def clean(self):
         if self.frequency in LEGACY_FREQUENCIES:
             self.frequency = LEGACY_FREQUENCIES[self.frequency]
 
+        resources_ids = set()
+        for resource in self.resources:
+            if resource.id in resources_ids:
+                raise MongoEngineValidationError(
+                    f"Duplicate resource ID {resource.id} in dataset #{self.id}."
+                )
+            resources_ids.add(resource.id)
```

Comment:
Could we do something like?
```
if len(set(self.resources)) != len(res.id for res in self.resources):
    raise...
```

Reply:
Sorry, I think I meant
```
if len(set(res.id for res in self.resources)) != len(self.resources):
    raise...
```

Reply:
Done in c69624788 (we just loose the `resource.id` in the exception) but it's simpler